Improve documentation

AurelienLP opened the pull request.

xlqian commented on README.md:
```diff
@@ -1,4 +1,64 @@
+
+## How to use
+
+To use Asgard with Navitia, you will need to give Jormungandr a configuration file telling Navitia to use Asgard for every street_network_modes you want. Here's a (not so) basic one:
```
```suggestion
To use Asgard with Navitia, you will need to give Jormungandr a configuration file telling Navitia to use Asgard for every street_network_modes you want. This file should be located in the `INSTANCE_DIR` which is set in the [default_settings.py](https://github.com/CanalTP/navitia/blob/dev/source/jormungandr/jormungandr/default_settings.py#L9) a (not so) basic one:
```

AurelienLP commented:
> I'll give general comments on the installation and run of asgard using this readme rather than the specific changes of this PR.
> 
>     * Build from source without docker : Ok
> 
>     * Build from source with docker : there is no folder "docker/asgard-dev", so I stopped there
> 
>     * Create data for asgard : the create_asgard_data.sh worked well
> 
>     * Run asgard : documentation is a bit missing and the executable has no --help. I needed to open the source code to understand a ASGARD_VALHALLA_CONF env variable is needed and should point to the valhalla.json generated by the script create_asgard_data.sh.
> 
>     * I tried to use the config file written in the readme and put it in INSTANCE_DIR folder for jormungandr as suggested by @xlqian , but it didn't work (seems to be on jormun's side).
>       It did work when I added to jormun .env file a line with
>       JORMUNGANDR_INSTANCE_TEST="content of the json file in this PR, but all on one line"
> 
> 
> Random comments :
> 
>     * Maybe put subsections inside the "From sources", like "build from source", "build in docker", "prepare data", "run unit tests", "install linters/formatter"
> 
>     * not really clear why there is two set of dockers in two different places (this repo and navitia-docker-compose)
> 
>     * the valhalla.json generated by create_asgard_data.sh has (hard-coded?) links to folders that does not exists

- "docker/asgard-dev" is the old name of the docker, this part was outdated so I removed it.
- I need to improve the documentation to run Asgard
- Asgard may need a --help
- Subsections added
- I should maybe documente create_asgard_data.sh a little more
- navitia-docker-compose just help running the dockers, I should be more precise on that
- I should check the Jormun part and be clearer
